diff --git a/src/polkit/polkit-context.c b/src/polkit/polkit-context.c
index 14d08f0..d1b4910 100644
--- a/src/polkit/polkit-context.c
+++ b/src/polkit/polkit-context.c
@@ -311,52 +311,25 @@ polkit_context_init (PolKitContext *pk_context, PolKitError **error)
 #else
         if (pk_context->io_add_watch_func != NULL) {
                 pk_context->inotify_fd = inotify_init ();
-                if (pk_context->inotify_fd < 0) {
-                        polkit_debug ("failed to initialize inotify: %s", strerror (errno));
-                        /* TODO: set error */
-                        goto error;
-                }
+                if (pk_context->inotify_fd >= 0) {
 
-                /* Watch the /etc/PolicyKit/PolicyKit.conf file */
-                pk_context->inotify_config_wd = inotify_add_watch (pk_context->inotify_fd, 
-                                                                   PACKAGE_SYSCONF_DIR "/PolicyKit/PolicyKit.conf", 
-                                                                   IN_MODIFY | IN_CREATE | IN_ATTRIB);
-                if (pk_context->inotify_config_wd < 0) {
-                        polkit_debug ("failed to add watch on file '" PACKAGE_SYSCONF_DIR "/PolicyKit/PolicyKit.conf': %s",
-                                   strerror (errno));
-                        /* TODO: set error */
-                        goto error;
-                }
-
-                /* Watch the /usr/share/PolicyKit/policy directory */
-                pk_context->inotify_policy_wd = inotify_add_watch (pk_context->inotify_fd, 
-                                                                   PACKAGE_DATA_DIR "/PolicyKit/policy", 
-                                                                   IN_MODIFY | IN_CREATE | IN_DELETE | IN_ATTRIB);
-                if (pk_context->inotify_policy_wd < 0) {
-                        polkit_debug ("failed to add watch on directory '" PACKAGE_DATA_DIR "/PolicyKit/policy': %s",
-                                   strerror (errno));
-                        /* TODO: set error */
-                        goto error;
-                }
+                    /* Watch the /etc/PolicyKit/PolicyKit.conf file */
+                    pk_context->inotify_config_wd = inotify_add_watch (pk_context->inotify_fd, 
+                                                                       PACKAGE_SYSCONF_DIR "/PolicyKit/PolicyKit.conf", 
+                                                                       IN_MODIFY | IN_CREATE | IN_ATTRIB);
 
+                    /* Watch the /usr/share/PolicyKit/policy directory */
+                    pk_context->inotify_policy_wd = inotify_add_watch (pk_context->inotify_fd, 
+                                                                       PACKAGE_DATA_DIR "/PolicyKit/policy", 
+                                                                       IN_MODIFY | IN_CREATE | IN_DELETE | IN_ATTRIB);
 #ifdef POLKIT_AUTHDB_DEFAULT
-                /* Watch the /var/lib/misc/PolicyKit.reload file */
-                pk_context->inotify_grant_perm_wd = inotify_add_watch (pk_context->inotify_fd, 
-                                                                       PACKAGE_LOCALSTATE_DIR "/lib/misc/PolicyKit.reload", 
-                                                                       IN_MODIFY | IN_CREATE | IN_ATTRIB);
-                if (pk_context->inotify_grant_perm_wd < 0) {
-                        polkit_debug ("failed to add watch on file '" PACKAGE_LOCALSTATE_DIR "/lib/misc/PolicyKit.reload': %s",
-                                   strerror (errno));
-                        /* TODO: set error */
-                        goto error;
-                }
+                    /* Watch the /var/lib/misc/PolicyKit.reload file */
+                    pk_context->inotify_grant_perm_wd = inotify_add_watch (pk_context->inotify_fd, 
+                                                                           PACKAGE_LOCALSTATE_DIR "/lib/misc/PolicyKit.reload", 
+                                                                           IN_MODIFY | IN_CREATE | IN_ATTRIB);
 #endif
 
-                pk_context->inotify_fd_watch_id = pk_context->io_add_watch_func (pk_context, pk_context->inotify_fd);
-                if (pk_context->inotify_fd_watch_id == 0) {
-                        polkit_debug ("failed to add io watch");
-                        /* TODO: set error */
-                        goto error;
+                    pk_context->inotify_fd_watch_id = pk_context->io_add_watch_func (pk_context, pk_context->inotify_fd);
                 }
         }
 #endif
