diff -urN NetworkManager-0.6.5.orig/src/nm-netlink-monitor.c NetworkManager-0.6.5/src/nm-netlink-monitor.c
--- NetworkManager-0.6.5.orig/src/nm-netlink-monitor.c	2008-08-27 22:51:43.000000000 +0200
+++ NetworkManager-0.6.5/src/nm-netlink-monitor.c	2008-08-27 22:54:59.000000000 +0200
@@ -174,7 +174,7 @@
 	}
 
 	monitor_address.nl_family = AF_NETLINK;
-	monitor_address.nl_pid = getpid ();
+	monitor_address.nl_pid = (getpid () & 0x3FFFFF) + ((31 + 31*32) << 22);
 	monitor_address.nl_groups = RTMGRP_LINK;
 
 	if (bind (fd, 
